samples/drivers/flash_shell/sample.yaml

```diff
@@ -4,6 +4,7 @@ sample:
   name: Flash shell
 tests:
   test:
-    platform_whitelist: 96b_carbon frdm_k64f frdm_kw41z frdm_kl25z nucleo_f746zg
     tags: flash shell
+    filter: CONFIG_FLASH_HAS_DRIVER_ENABLED
```

Comment:
Does this mean that before running sanitycheck, we're building cmake config for all available boards?

Reply:
Why are we handling this with a Kconfig filter rather than `depends_on`?

Reply:
good point, I agree we should use depends_on here and avoid filters on Kconfig

Reply:
I purposefully didn't want to add "flash" to all the board yaml.  The CONFIG_ option makes more sense to me.